psycopg3/types/datetime.pyx

```diff
@@ -0,0 +1,17 @@
+"""
+Cython adapters for datetime types.
+"""
+
+# Copyright (C) 2020 The Psycopg Team
+
+
+from libc.stdint cimport *
+
+from psycopg3.types.text cimport load_text, get_context_text
+
+
+cdef void register_text_c_loaders():
+    logger.debug("registering optimised datetime c loaders")
+    from psycopg3.types import datetime
+    register_c_loader(datetime.DateTimeLoader.load, load_text, get_context_text)
```

Comment:
I'd rather these to be undefined than set to the wrong ones. Probably better to work on the python side of it and then write an optimised implementation.